addons/purchase_stock/tests/test_purchase_order_process.py

```diff
@@ -57,3 +59,40 @@ def test_01_packaging_propagation(self):
         })
         po.button_confirm()
         self.assertEqual(po.order_line.move_ids.product_packaging_id, packaging)
+
+    def test_02_vendor_delay_report_partially_cancelled_purchase_order(self):
+        """ Test vendor delay reports for partially cancelled purchase order"""
+        partner = self.partner_1
+        purchase_order = self.env['purchase.order'].create({
+            'partner_id': partner.id,
+            'date_planned': fields.Datetime.now().replace(hour=23, minute=0) + timedelta(days=2),
+            'order_line': [
+                Command.create({
+                    'product_id': self.product_1.id,
+                    'product_qty': 2.0,
+                    'product_uom': self.product_1.uom_id.id,
+                }),
+                Command.create({
+                    'product_id': self.product_2.id,
+                    'product_qty': 3.0,
+                    'product_uom': self.product_2.uom_id.id,
+                })],
+        })
+        purchase_order.button_confirm()
+        purchase_order.picking_ids.move_ids.flush_recordset()
```

Comment:
Do you really need those `flush_recorset()`? :thinking:

Reply:
@clesgow at least the runbot doesn't pass the test without the `flush_recordset()`. I didn't have any trouble in local though.

Reply:
'kay so that was pretty weird to me and asked a bit further. Turns out there should a `_depends` defined on the report model `vendor.delay.report` itself that takes care of flushing the right fields.
Something like that:
```py
_depends = {
    'product.product': ['product_tmpl_id'],
    'product.template': ['uom_id', 'categ_id'],
    'purchase.order': ['partner_id'],
    'purchase.order.line': ['date_planned', 'order_id'],
    'stock.move': ['date', 'purchase_line_id', 'product_id'],
    'stock.move.line': ['qty_done', 'move_id', 'product_uom_id'],
    'uom.uom': ['factor'],
}
```
It *should* solve the `flush_recordset()` issue in a more proper, so do you mind trying that instead? :sweat_smile: